src/log.rs

```diff
@@ -17,3 +20,49 @@ pub const DEBUG: tracing::Level = tracing::Level::DEBUG;
 pub const INFO: tracing::Level = tracing::Level::INFO;
 #[cfg(feature = "tracing")]
 pub const WARN: tracing::Level = tracing::Level::WARN;
+
+///A record containing all of the logging metadata from librdkafka
+#[derive(Debug, Clone)]
+pub struct LogRecord {
+    level: RDKafkaLogLevel,
+    fac: String,
+    log_message: String,
+    contexts: String,
+}
+
+impl LogRecord {
+    /// Build a new LogRecord
+    pub fn new(level: RDKafkaLogLevel, fac: String, log_message: String, contexts: String) -> Self {
+        LogRecord {
+            level,
+            fac,
+            log_message,
+            contexts,
+        }
+    }
+
+    /// The librdkafka log level for this record
+    pub fn level(&self) -> RDKafkaLogLevel {
+        self.level
+    }
+
+    /// The librdkafka facility for this record
+    pub fn facility(&self) -> &str {
+        &self.fac
+    }
+
+    /// The message for this record
+    pub fn log_message(&self) -> &str {
+        &self.log_message
+    }
+
+    /// The set of debug contexts for this record
+    pub fn contexts(&self) -> &str {
+        &self.contexts
+    }
+
+    /// An iterator over the CSV context items
+    pub fn split_contexts(&self) -> Split<'_, &str> {
```

Comment:
Better to have this be `impl Iterator<Item = &str>`, I think? Gives you cover to change it in the future if need be.

Reply:
Fixed in d62c66a019e8036257495df8f9f5758faeb8ed84.